### PR TITLE
Use runno-clang for C/C++

### PR DESCRIPTION
**Pull request by LukePrior**
I think this solves #193

### CODE REVIEW

File: packages/runtime/src/provider.ts [hunk 1]
```diff
@@ -34,17 +34,7 @@ function commandsForRuntime(name: string, entryPath: string): RuntimeCommands {
     return { run: `cat ${entryPath} | sqlite` };
   }
 
-  if (name === "clang") {
-    return {
-      prepare: [
-        `clang -cc1 -Werror -triple wasm32-unkown-wasi -isysroot /sys -internal-isystem /sys/include -ferror-limit 4 -fmessage-length 80 -fcolor-diagnostics -O2 -emit-obj -o ./program.o ${entryPath}`,
```
Comment (taybenlor) [inline packages/runtime/src/provider.ts:40]: Instead of removing the `clang` runtime I'd like to see `clang` here switched for `runno-clang`.

**Comment by taybenlor** [comment]
I'd be happy to try and help you get the dev environment running locally! It's probably got some quirks because I've been devving it by myself.

How did you go following the instructions in the `README.md`?

**Review by taybenlor — CHANGES_REQUESTED** [review]
Similar issue to before, now you're using the C++ args for the C version.

File: packages/runtime/src/provider.ts [hunk 2]
```diff
@@ -37,8 +37,8 @@ function commandsForRuntime(name: string, entryPath: string): RuntimeCommands {
   if (name === "clang") {
     return {
       prepare: [
-        `clang -cc1 -Werror -triple wasm32-unkown-wasi -isysroot /sys -internal-isystem /sys/include -ferror-limit 4 -fmessage-length 80 -fcolor-diagnostics -O2 -emit-obj -o ./program.o ${entryPath}`,
-        `wasm-ld -L/sys/lib/wasm32-wasi /sys/lib/wasm32-wasi/crt1.o ./program.o -lc -o ./program.wasm`,
+        `runno-clang -cc1 -Werror -emit-obj -disable-free -isysroot /sys -internal-isystem /sys/include/c++/v1 -internal-isystem /sys/include -internal-isystem /sys/lib/clang/8.0.1/include -ferror-limit 4 -fmessage-length 80 -fcolor-diagnostics -O2 -o program.o -x c++  ${entryPath}`,
+        `runno-wasm-ld --no-threads --export-dynamic -z stack-size=1048576 -L/sys/lib/wasm32-wasi /sys/lib/wasm32-wasi/crt1.o program.o -lc -lc++ -lc++abi -o ./program.wasm`,
```
[the remark this answers is not in the document]
Comment (LukePrior) [reply packages/runtime/src/provider.ts]: Sorry, that makes perfect sense, I _think_ now it's right.

**Comment by taybenlor** [comment]
Awesome, this looks right. I'm just going to test it works locally.

**Review by taybenlor — CHANGES_REQUESTED** [review]
Alright, annoyingly this doesn't seem to work. Here's the output I get locally for your example:

```
In file included from program:1:
In file included from /sys/include/stdlib.h:6:
In file included from /sys/include/__functions_malloc.h:7:
/sys/include/stddef.h:30:15: fatal error: 'stddef.h' file not found
#include_next <stddef.h>
              ^~~~~~~~~~
1 error generated.
```

My clang knowledge is pretty limited and this setup is a bit odd. I think you'll have to get it running locally on your machine to start debugging it.

Could you go through the instructions in the `README.md` under `## Running locally` to get it going?

Currently you'll need to run `npm run dev` separately inside each of the `client` and `website` directories. When you modify the `runtime` you'll need to run build again.

If you have any issues with that comment here and I'll help you out.

I'll work on improving the local dev experience as well.

**Comment by taybenlor** [comment]
If you'd like to organise a pairing session so we can work through it together I'd also be happy to do that. Just email `runno@taybenlor.com` and we can coordinate a video call during the week (12th or 13th will be best for me).

**Comment by LukePrior** [comment]
I will have a look into this further when I have some time, not sure when so I will close this for now.

**Pull request closed without merging**